### PR TITLE
Fix checkbox editor widget against datasets delivering boolean fields as strings

### PR DESCRIPTION
**Pull request by nirvn**
Fix for a regression seen running against QGIS master.

### CODE REVIEW

**Comment by nirvn** [comment]
@suricactus , if you can think of a better way to detect whether value is a string other than by checking for presence of the toLowerCase function, I'm all ears :)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9f32d578f619ffbbbc855aeb600c8adba46320fe: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4597-4330b03b7667d2e011bdf252c7abcab9f25e432b-x86-android.apk)
</details>

File: src/qml/editorwidgets/CheckBox.qml [hunk 1]
```diff
@@ -57,7 +57,12 @@ EditorWidgetBase {
 
     checked: {
         if (isBool) {
-            return !isNull ? value : false;
+            var actualValue = value
+            // Some datasets - such as Geopackage - can send the value as a 'True' or 'False' string, we have to work around that
+            if (!isNull && value.toLowerCase !== undefined) {
```
Comment (suricactus) [inline src/qml/editorwidgets/CheckBox.qml:62]:
```suggestion
            if (!isNull && typeof value == 'string') {
```

This does not work?

**Review by suricactus — APPROVED** [review]
Besides the weird check the code looks good. Have not checked functionally what it fixed.

File: src/qml/editorwidgets/CheckBox.qml (hunk 1 above)
Comment (nirvn) [reply src/qml/editorwidgets/CheckBox.qml]: Yep, thanks.

**Pull request merged**